### ISSUE TITLE
UUID/NullUUID Scan Panic v1.1.10 (but not v.1.1.9) - Postgres

### ISSUE THREAD

**Issue by jpfluger** [uptrace/bun#760]
The scan error happens when the **first** column to be scanned is a `null` UUID (using type NullUUID from `github.com/gofrs/uuid`). If the first column scanned is replaced by UUID (rather than NullUUID) and that value is valid, then columns of NullUUID are processed normally. 

The problem began in v1.1.10 but not v.1.1.9.  The work-around is to either (a) downgrade OR to (b) ensure the first column has a valid UUID value.

Here are the tests that I ran that you can use for proofs. The scan for `MyTest1` **will fail** but the scan for MyTest2 **will succeed** :

```go
type MyTest1 struct {
	F1_NullUUID uuid.NullUUID `bun:"f1,type:uuid"`
	F2_UUID     uuid.UUID     `bun:"f2,type:uuid"`
	F3_String   string        `bun:"f3,type:text"`
	F4_Bool     bool          `bun:"f4,notnull,type:boolean"`
	F5_Time     *time.Time    `bun:"f5,nullzero,type:timestamp"`
}
type MyTest1s []*MyTest1

type MyTest2 struct {
	F2_UUID     uuid.UUID     `bun:"f2,type:uuid"`
	F1_NullUUID uuid.NullUUID `bun:"f1,type:uuid"`
	F3_String   string        `bun:"f3,type:text"`
	F4_Bool     bool          `bun:"f4,notnull,type:boolean"`
	F5_Time     *time.Time    `bun:"f5,nullzero,type:timestamp"`
}
type MyTest2s []*MyTest2

func TestQueries(db *bun.DB) error {

	const myTest2 = `SELECT '7d4b8e55-890b-4a33-b516-3123411d1f0a' as "f2", null as "f1", 'some-text' as "f3", true as "f4", timestamptz('2023-01-18 17:36:22.794035+00') as "f5";`

	mts2 := MyTest2s{}
	q := db.NewRaw(myTest2)

       // QUERY SUCCEEDS !!!
	if err := q.Scan(context.Background(), &mts2); err != nil {
		return err
	}

	const myTest1 = `SELECT null as "f1", '7d4b8e55-890b-4a33-b516-3123411d1f0a' as "f2", 'some-text' as "f3", true as "f4", timestamptz('2023-01-18 17:36:22.794035+00') as "f5";`

	mts1 := MyTest1s{}
	q = db.NewRaw(myTest1)

       // QUERY WILL PANIC !!!
	if err := q.Scan(context.Background(), &mts1); err != nil {
		return nil, err
	}

        return nil
}
```

Proof that MyTest2 succeeds: 

![image](https://user-images.githubusercontent.com/6799005/213256955-9473430e-1615-4090-aa21-7d6eade227c5.png)

This is the error from the panic on the scan in MyTest1:

![image](https://user-images.githubusercontent.com/6799005/213257207-8570c703-fc8b-45fb-b573-cd74a13ff460.png)

This is the stack from the panic on MyTest1:

```
Goroutine 4313 [running]:
github.com/labstack/echo/v4/middleware.RecoverWithConfig.func1.1.1()
	/drive2/my-projects/goland/pkg/mod/github.com/labstack/echo/v4@v4.10.0/middleware/recover.go:93 +0x2b3
panic({0x1585e40, 0xc0000e71d0})
	/opt/go/active/src/runtime/panic.go:890 +0x267
reflect.Value.Field({0x15242e0, 0xc0020be730, 0x16}, 0x0)
	/opt/go/active/src/reflect/value.go:1266 +0x26f
github.com/uptrace/bun/schema.fieldByIndex({0x15242e0, 0xc0020be730, 0x16}, {0xc000718390, 0x1, 0x1})
	/drive2/my-projects/goland/pkg/mod/github.com/uptrace/bun@v1.1.10/schema/reflect.go:33 +0xc5
github.com/uptrace/bun/schema.(*Field).ScanValue(0xc0006151e0, {0x15242e0, 0xc0020be730, 0x16}, {0x0, 0x0})
	/drive2/my-projects/goland/pkg/mod/github.com/uptrace/bun@v1.1.10/schema/field.go:114 +0x18f
github.com/uptrace/bun.(*structTableModel).scanColumn(0xc0003ac780, {0xc00085f400, 0x2}, {0x0, 0x0})
	/drive2/my-projects/goland/pkg/mod/github.com/uptrace/bun@v1.1.10/model_table_struct.go:331 +0x1dc
github.com/uptrace/bun.(*structTableModel).ScanColumn(0xc0003ac780, {0xc00085f400, 0x2}, {0x0, 0x0})
	/drive2/my-projects/goland/pkg/mod/github.com/uptrace/bun@v1.1.10/model_table_struct.go:311 +0x87
github.com/uptrace/bun.(*structTableModel).Scan(0xc0003ac780, {0x0, 0x0})
	/drive2/my-projects/goland/pkg/mod/github.com/uptrace/bun@v1.1.10/model_table_struct.go:307 +0xdf
database/sql.convertAssignRows({0x1691320, 0xc0003ac780}, {0x0, 0x0}, 0xc000465f00)
	/opt/go/active/src/database/sql/convert.go:385 +0x1ef9
database/sql.(*Rows).Scan(0xc000465f00, {0xc0020be640, 0x5, 0x5})
	/opt/go/active/src/database/sql/sql.go:3287 +0x4ae
github.com/uptrace/bun.(*structTableModel).scanRow(0xc0003ac780, {0x2000750, 0xc0000ec000}, 0xc000465f00, {0xc0020be640, 0x5, 0x5})
	/drive2/my-projects/goland/pkg/mod/github.com/uptrace/bun@v1.1.10/model_table_struct.go:292 +0xc8
github.com/uptrace/bun.(*sliceTableModel).ScanRows(0xc0003ac780, {0x2000750, 0xc0000ec000}, 0xc000465f00)
	/drive2/my-projects/goland/pkg/mod/github.com/uptrace/bun@v1.1.10/model_table_slice.go:73 +0x425
github.com/uptrace/bun.(*baseQuery).scan(0xc0003ac690, {0x2000750, 0xc0000ec000}, {0x20012e8, 0xc0003ac690}, {0xc0029501a0, 0x158}, {0x1ffb888, 0xc0003ac780}, 0x1)
	/drive2/my-projects/goland/pkg/mod/github.com/uptrace/bun@v1.1.10/query_base.go:568 +0x37b
github.com/uptrace/bun.(*RawQuery).Scan(0xc0003ac690, {0x2000750, 0xc0000ec000}, {0xc002226630, 0x1, 0x1})
	/drive2/my-projects/goland/pkg/mod/github.com/uptrace/bun@v1.1.10/query_raw.go:60 +0x275
MY FUNCTIONS TRACED HERE...
```

**Comment by siketyan** [comment]
Same here.
We have fields such as `*UpdatedAt` which is nullable, implements `sql.Scanner` and `driver.Valuer`. Non-null fields do NOT cause this problem.
Also we confirmed the problem occur on both INSERT and SELECT queries.

**Comment by jpfluger** [comment]
Thank you @siketyan and @vmihailenco . This is fixed in my tests using v1.1.12.  Related to #776 and #786.
